### PR TITLE
[#13] As a user, I can view the list of keywords and their results uploaded by CSV files

### LINKED ISSUE #13: [UI] As a user, I can view the list of keywords and their results uploaded by CSV files
## Why

User needs to see all the records from their previously uploaded CSV

## Who Benefits?

The user who needed to use view and filter the results from uploaded CSV

### PR DESCRIPTION
**Pull request by nvminhtue**
close #13 

## What happened

Show the list of scraping keywords and their own result.
 
## Insight

- Implement UI of keywords list
- Implement UI of keyword result
- Implement the keyword's HTML result
- Add event on clicking on Logo button which returns to home page
- Adjust prettier convention on JSON files
 
## Proof Of Work

Show us the implementation: screenshots, gif, etc.
image

image

image

### CODE REVIEW

File: lib/elixir_internal_certificate_web/templates/user_search/_keyword_list.html.heex [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    </tr>
+  </thead>
+  <tbody>
+    <tr role="button" class="align-middle">
+      <td scope="row">1</td>
+      <td>Mark</td>
+      <td class="text-success">Success</td>
+      <td>
+        <%=
+          button "Html response",
```
Comment (andyduong1920) [inline lib/elixir_internal_certificate_web/templates/user_search/_keyword_list.html.heex]: small comment we can use the `link` https://hexdocs.pm/phoenix_html/Phoenix.HTML.Link.html#link/2 for the `get link`

File: lib/elixir_internal_certificate_web/templates/user_search/_upload_file.html.heex [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<.form id="keyword-upload-form" let={f} for={@conn} action={Routes.user_search_path(@conn, :upload)} multipart={true}>
+  <%= label f, :File, "Select CSV file, maximum 1000 keywords contained", class: "form-label" %>
```
Comment (andyduong1920) [inline lib/elixir_internal_certificate_web/templates/user_search/_upload_file.html.heex]:
```suggestion
  <%= label f, :file, "Select CSV file, maximum 1000 keywords contained", class: "form-label" %>
```

File: lib/elixir_internal_certificate_web/templates/user_search/index.html.heex [hunk 3]
```diff
@@ -1,13 +1,4 @@
...
-    <div class="d-flex me-4">
-      <%= file_input f, :file, accept: ".csv" , required: true, class: "form-control h-100 w-50 #{error_class(f, :file)}" %>
-      <%= error_tag f, :file %>
-
-      <div class="w-100">
-        <%= submit "Upload", class: "btn btn-primary btn-block ms-2" %>
-      </div>
-    </div>
-  </.form>
+  <%= render "_upload_file.html", assigns %>
```
Comment (andyduong1920) [inline lib/elixir_internal_certificate_web/templates/user_search/index.html.heex]: How about 

```suggestion
  <%= render "_upload_form.html", assigns %>
```

File: test/elixir_internal_certificate_web/features/search_list/searching_list_test.exs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+defmodule ElixirInternalCertificateWeb.SearchList.SearchingListTest do
```
Comment (andyduong1920) [inline test/elixir_internal_certificate_web/features/search_list/searching_list_test.exs]: is it a search feature, can we name it as just `view_list_keyword`

File: lib/elixir_internal_certificate_web/templates/user_search/_keyword_list.html.heex (hunk 1 above)
Comment (nvminhtue) [reply lib/elixir_internal_certificate_web/templates/user_search/_keyword_list.html.heex]: Good catch, updated in [1889475](https://github.com/nvminhtue/elixir_internal_certificate/pull/36/commits/1889475a2a989200399e58c18afa287a31772e39)

File: test/elixir_internal_certificate_web/features/search_list/searching_list_test.exs (hunk 4 above)
Comment (nvminhtue) [reply test/elixir_internal_certificate_web/features/search_list/searching_list_test.exs]: updated to `view_keyword_list` module in [1889475](https://github.com/nvminhtue/elixir_internal_certificate/pull/36/commits/1889475a2a989200399e58c18afa287a31772e39)

**Review by andyduong1920 — APPROVED** [review]

**Review by junan — APPROVED** [review]

File: lib/elixir_internal_certificate_web/controllers/user_search_controller.ex [hunk 5]
```diff
@@ -5,7 +5,13 @@ defmodule ElixirInternalCertificateWeb.UserSearchController do
   alias ElixirInternalCertificateWeb.CsvParsingHelper
   alias ElixirInternalCertificateWeb.Router.Helpers, as: Routes
 
-  def index(conn, _params), do: render(conn, "index.html")
```
Comment (junan) [inline lib/elixir_internal_certificate_web/controllers/user_search_controller.ex:8]: Why we changing it from  1 liner syntax to `do/end`?

**Pull request merged**